torch/_utils.py

```diff
@@ -935,3 +938,25 @@ def queue_seed(self, cb, traceback):
 
     def get_calls(self) -> List:
         return self.call_order
+
+
+logger = logging.getLogger(__name__)
+P = ParamSpec("P")
+
+
+class CallbackRegistry(Generic[P]):
```

Comment:
Move `CallbackRegistry` from `torch.utils._cuda_trace.py` to here, which can be shared by each backend.